markdownify/__init__.py

```diff
@@ -170,10 +170,11 @@ def process_tag(self, node, convert_as_inline, children_only=False):
                 newlines = '\n' * max(newlines_left, newlines_right)
                 text = text_strip + newlines + next_text_strip
 
-        if not children_only:
-            convert_fn = getattr(self, 'convert_%s' % node.name, None)
-            if convert_fn and self.should_convert_tag(node.name):
-                text = convert_fn(node, text, convert_as_inline)
+        # apply this tag's final conversion function
+        # (BeautifulSoup objects have a name of "[document]" and will not apply one)
```

Comment:
Maybe the comment would be clearer for future reference if it read:

> BeautifulSoup's special objects (e.g. the root object) have special names (like `[document]`) and do not match any `convert_*` functions. These objects therefore get skipped automatically by the next part.

Reply:
I'm glad you mentioned this! If [this suggestion in #136](https://github.com/matthewwithanm/python-markdownify/pull/136#issuecomment-2567101210) is implemented, then indeed we could support final conversion functions on `BeautifulSoup` objects! This could be useful for inserting YAML metadata, or auto-applying a final cleanup via regex, and other fun things like that.